src/common/class-syntax.k

```diff
@@ -0,0 +1,125 @@
+
+syntax ConstrInv ::= TypeArgs "this"  "(" Exps ")" ";"        [prefer, klabel('AltConstrInv)]
+syntax ConstrInv ::= "this"  "(" Exps ")" ";"                 [klabel('AltConstrInv)]
+syntax ConstrInv ::= TypeArgs "super"  "(" Exps ")" ";"       [prefer, klabel('SuperConstrInv)]
+syntax ConstrInv ::= "super"  "(" Exps ")" ";"                [klabel('SuperConstrInv)]
+syntax ConstrInv ::= Exp "." TypeArgs "super"  "(" Exps ")" ";" [prefer, klabel('QSuperConstrInv)]
+syntax ConstrInv ::= Exp "." "super"  "(" Exps ")" ";"          [klabel('QSuperConstrInv)]
+
+syntax ConstrMod ::= Public
+                  | Private
+                  | Protected
+syntax AnnoConstrMod ::= Anno | ConstrMod
+
+//@ \subsection{EnumDeclarations.sdf}
+syntax EnumDecHead ::= AnnoClassModList "enum" Id Interfaces [prefer, klabel('EnumDecHead)]
+                     | AnnoClassModList "enum" Id            [klabel('EnumDecHead)]
+
+syntax EnumDec ::= EnumDecHead EnumBody                     [klabel('EnumDec)]
+
+syntax EnumBody ::= "{" EnumConstList EnumBodyDecs"}"       [prefer, klabel('EnumBody)]
+                  | "{" EnumConstList "}"                   [klabel('EnumBody)]
+                  | "{" EnumConstList "," EnumBodyDecs "}"  [prefer, klabel('EnumBody)]
+                  | "{" EnumConstList "," "}"               [klabel('EnumBody)]
+//todo optionals
+syntax EnumConst ::= Id EnumConstArgs ClassBody [prefer, klabel('EnumConst)]
+
+syntax EnumConstArgs ::= "(" Exps ")"           [bracket]
+
+syntax EnumBodyDecs ::= ";" ClassBodyDecList    [klabel('EnumBodyDecs)]
+
+//@ \subsection{FieldDeclarations.sdf}
+syntax FieldDec ::= AnnoFieldModList Type VarDecList ";"    [klabel('FieldDec)]
+
+syntax VarDec ::= VarDecId                [klabel('VarDec)]
+               // | VarDecId "=" VarInit    [klabel('VarDec)]
+
+syntax VarDecId ::= Id | Id Dims          [klabel('ArrayVarDecId)]
+
+syntax VarInit ::= Exp | ArrayInit
+
+syntax FieldMod ::= Public
+                  | Private
+                  | Protected
+                  | Final
+                  | Static
+                  | Transient
+                  | Volatile
+syntax AnnoFieldMod ::= Anno | FieldMod
+
+//@ \subsection{MethodDeclarations.sdf}
+syntax Param ::= AnnoVarModList Type VarDecId           [klabel('Param)]
+syntax Param ::= AnnoVarModList Type "..." VarDecId     [klabel('VarArityParam)]
+
+syntax Throws ::= "throws" ExceptionTypeList
+syntax ExceptionType ::= ClassType
+syntax MethodBody ::= Block | ";" [klabel('NoMethodBody)]
+//todo optionals
+syntax MethodDecHead ::= AnnoMethodModList TypeParams Type Id "(" Params ")" Throws [prefer, klabel('MethodDecHead)]
+                       | AnnoMethodModList TypeParams Type Id "(" Params ")"    [klabel('MethodDecHead)]
+                       | AnnoMethodModList Type Id "(" Params ")" Throws   [klabel('MethodDecHead)]
+                       | AnnoMethodModList Type Id "(" Params ")"    [klabel('MethodDecHead)]
```

Comment:
Same here.  We would need 4 different macros here to desugar the options into the general construct.  Letting JavaFront do that for us is just prolonging the inevitable.  Andrei, Brandon, Radu, are you also watching this repo?